### PR TITLE
Minimal list support

### PR DESCRIPTION
**Pull request by StephanMeijer**
Emit `StartOrderedListItem`/`EndOrderedListItem` and `StartUnorderedListItem`/`EndUnorderedListItem` for DOCX paragraphs with `<w:numPr>`.

Ordered styles: Decimal, LowerAlpha, UpperAlpha, LowerRoman, UpperRoman. Unordered: Disc.

## Summary by CodeRabbit

* **New Features**
  * Streaming support for ordered and unordered DOCX lists: nested list items, per-level classification, start-number behavior, emitted start/end list-item events, and proper closure at table/cell boundaries and document end.

* **Documentation**
  * README and crate docs updated to describe supported list behavior and a “Lists (V1 cuts)” section enumerating intentional V1 limitations.

* **Tests**
  * Extensive unit and integration tests covering numbering parsing, list emission, edge cases, boundaries, and cross-feature interactions.

### CODE REVIEW

File: crates/docspec-docx-reader/src/lib.rs [hunk 1]
```diff
@@ -35,13 +37,19 @@
...
 //!   fields, `<w:tcPr>` visual fields, `<w:tblGrid>`)
-//! - Lists
 //! - Drawings and images (`<w:drawing>`, `<w:pict>`)
 //! - Comments, footnotes, headers, footers
 //! - Document metadata
 //! - Tracked deletions (`<w:del>`, `<w:moveFrom>`) — accept-changes semantics
 //! - Structured document tag properties (`<w:sdtPr>`, `<w:sdtEndPr>`)
 //!
+//! # List Support (V1)
+//!
```
Comment (StephanMeijer) [inline crates/docspec-docx-reader/src/lib.rs]:
```suggestion
```

Comment (StephanMeijer) [reply crates/docspec-docx-reader/src/lib.rs]: TLDR'd: section is now a single line pointing at the crate README (which already documents V1 list semantics and the V1 cuts). Amended into 95e1cb2.

**Comment by StephanMeijer** [comment]
@kody review

**Comment by StephanMeijer** [comment]
@kody review

**Comment by StephanMeijer** [comment]
@kody review

File: crates/docspec-docx-reader/src/document.rs [hunk 2]
```diff
@@ -302,12 +359,94 @@ impl DocumentReader {
...
+                docspec_core::ListStyleType::Disc
+            };
+            let phantom_start = self.compute_start(num_id);
+            let start_event = if is_ordered {
+                Event::StartOrderedListItem {
+                    id: Some(num_id.to_string()),
+                    level: phantom_ilvl,
+                    start: phantom_start,
+                    style_type: phantom_style,
+                }
```
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec-docx-reader/src/document.rs]: Good catch on the semantic placement. Fixed in 2d03869: phantom levels in `reconcile_list_stack` no longer call `compute_start`, so `start: Some(1)` now lands on the user-authored item at the target ilvl rather than the synthetic phantom at level 0. Updated the two existing tests (`ilvl_overflow_clamps_to_eight` in both `src/document.rs` and `tests/reader.rs`) and added a focused regression test `first_list_item_at_deep_level_keeps_start_marker_on_real_item`.

Pushing back gently on the "high severity" classification though — in this V1 `start` is only ever `Some(1)` (no `<w:start>` parsing yet, per the README's V1 cuts), so a reasonable writer that defaults empty ordered lists to start-at-1 produces identical rendered output under either placement. The fix is the correct contract and future-safe for when `<w:start>` lands, but it's not currently producing wrong output.

**Comment by StephanMeijer** [comment]
@kody review

**Comment by StephanMeijer** [comment]
@kody review

File: crates/docspec-docx-reader/src/document.rs [hunk 3]
```diff
@@ -302,12 +359,93 @@ impl DocumentReader {
...
+                        Event::EndUnorderedListItem
+                    };
+                    self.queue.push_back(end);
+                    break;
+                }
+                core::cmp::Ordering::Less => break,
+            }
+        }
+
+        let target_depth = usize::try_from(ilvl).unwrap_or(usize::MAX);
```
[the remark this answers is not in the document]
Comment (StephanMeijer) [reply crates/docspec-docx-reader/src/document.rs:473]: Intentional V1 cut. Phantom intermediate levels (synthesised when the first authored item appears at `ilvl > 0`) inherit the target item's ordered/unordered classification and use `Decimal`/`Disc` defaults. Real-world DOCX rarely starts a list at a deep `ilvl` with a mixed-format abstractNum, so the correctness gap is narrow and not worth the complexity in V1. Future work will resolve per-level `numFmt` from `MinimalNumbering::resolve` for each phantom level.
